M2/autogen.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+echo "-- Generating configure script"
+autoreconf --verbose --force --install
+
+echo "-- Updating M2-emacs submodule"
+git submodule update --init Macaulay2/editors/emacs
```

Comment:
Why just this one, as opposed to all of them?

Reply:
We may not need the others.  In the autotools build, the submodules that `configure` tells us we need to build M2 are updated in the corresponding Makefile in `M2/libraries/`.

Reply:
It would.  But so would all the other submodule stuff!

(In the Debian package, where we do build from a tarfile, this step is skipped entirely and a different Debian-specific program -- `dh_autoreconf` -- is run to generate the configure script.)

Reply:
I think since this script is just two commands, it would be worth checking if `../.git` exists (either as a directory or file in case of a worktree) and if not skipping the second command and giving a warning that some other tarfiles may need to be downloaded.

Reply:
Thinking about this some more, I don't think a warning is sufficient.  If we're not working in a clone of the git repository, then the build will certainly fail because we rely on git submodules, and we don't have a good way of fetching them without git.  I'll add an error message.

It might be good to distribute a tarball (probably generated using `git archive`) that includes all the submodules to support git-less builds.  The configure script could be pre-generated (which is standard for autotools projects) so that users wouldn't need to run this script.

Reply:
But if someone has done this already, why give an error and fail?